ts/features/itwallet/common/hooks/useItwDiscoveryBannerType.ts

```diff
@@ -1,9 +1,47 @@
+import { useEffect, useState } from "react";
+import * as O from "fp-ts/Option";
+import { pipe } from "fp-ts/function";
+import { itwIsWalletInstanceRemotelyActiveSelector } from "../store/selectors/preferences";
+import { useIOSelector } from "../../../../store/hooks.ts";
+
 type DiscoveryBannerType = "onboarding" | "reactivating";
 
 /**
- * Hook to determine the type of the ITW discovery banner to display based on the wallet's state
+ * Hook to determine the type of the ITW discovery banner to display based on the wallet's state.
+ * Returns "reactivating" when there's an active wallet instance that can be reactivated,
+ * otherwise returns "onboarding".
+ *
  * @returns the type of the banner to display
  */
-export const useItwDiscoveryBannerType = (): DiscoveryBannerType =>
-  // FIXME: (SIW-2048) we need to return the correct banner type based on the wallet's remote state
-  "onboarding";
+export const useItwDiscoveryBannerType = ():
+  | DiscoveryBannerType
+  | undefined => {
+  // Get the wallet instance status from Redux
+  const isWalletInstanceRemotelyActive = useIOSelector(
+    itwIsWalletInstanceRemotelyActiveSelector
+  );
+
+  // Store the determined banner type
+  const [determinedBannerType, setDeterminedBannerType] = useState<
+    DiscoveryBannerType | undefined
+  >();
+
+  useEffect(() => {
+    pipe(
+      O.fromNullable(isWalletInstanceRemotelyActive),
+      O.fold(
+        // If the value is still undefined, do nothing
+        () => undefined,
+        // Otherwise, set the banner type based on the definitive value
+        isActive => {
+          const newType: DiscoveryBannerType = isActive
+            ? "reactivating"
+            : "onboarding";
+          setDeterminedBannerType(newType);
+        }
+      )
+    );
+  }, [isWalletInstanceRemotelyActive]);
+
+  return determinedBannerType;
```

Comment:
We can remove the need to use state by returning immediately the result. What do you think?
```suggestion
  return pipe(
      O.fromNullable(isWalletInstanceRemotelyActive),
      O.fold(
        // If the value is still undefined, do nothing
        () => undefined,
        // Otherwise, set the banner type based on the definitive value
        isActive =>  isActive
            ? "reactivating"
            : "onboarding";
      )
    );
```

Reply:
Thanks for the suggestion, addressed here https://github.com/pagopa/io-app/pull/6889/commits/b3ce09a059e4b6b09893ffbdcac7a93098e62578!